File: src/main/java/org/fountainmc/api/entity/hanging/Painting.java
package org.fountainmc.api.entity.hanging;

import org.fountainmc.api.NonnullByDefault;
import org.fountainmc.api.entity.data.EntityData;
import org.fountainmc.api.entity.data.hanging.PaintingData;

/**
 * A painting that is hanging on a wall.
 */
@NonnullByDefault
public interface Painting extends PaintingData, HangingEntity {

    /**
     * Copy all of the given data to this entity.
     * <p>Doesn't copy passenger information.</p>
     *
     * @param data the data to copy from
     */
    @Override
    default void copyDataFrom(EntityData data) {
        PaintingData.super.copyDataFrom(data);
    }

    /**
     * Take a snapshot of this entity's data
     * <p>The resulting snapshot is thread-safe.</p>
     *
     * @return a snapshot
     */
    @Override
    PaintingData snapshot();
}

File: src/main/kotlin/org/fountainmc/api/entity/hanging/Painting.kt
package org.fountainmc.api.entity.hanging

import org.fountainmc.api.entity.Entity
import org.fountainmc.api.entity.data.EntityData
import org.fountainmc.api.entity.data.hanging.PaintingData

/**
 * A painting that is hanging on a wall.
 */
interface Painting : PaintingData, HangingEntity, Entity {

    /**
     * Copy all of the mutable properties from the given data into this object.
     *
     * @param data the data to copy from
     * @see Entity.copyDataFrom
     * @see PaintingData.copyDataFrom
     */
    override fun copyDataFrom(data: EntityData) {
        super<PaintingData>.copyDataFrom(data)
    }

    /**
     * Take a snapshot of this entity's data
     *
     * The resulting snapshot is thread-safe.
     *
     * @return a snapshot
     */
    override fun snapshot(): PaintingData
}
